libs/agno/agno/eval/reliability.py

```diff
@@ -0,0 +1,136 @@
+
+if TYPE_CHECKING:
+    from rich.console import Console
+
+from agno.agent.agent import Agent
+from agno.run.response import RunResponse
+from agno.utils.log import logger, set_log_level_to_debug, set_log_level_to_info
+
+
+@dataclass
+class ReliabilityResult:
+    eval_status: str
+    failed_tool_calls: List[str]
+    passed_tool_calls: List[str]
+
+    def print_eval(self, console: Optional["Console"] = None):
+        from rich.console import Console
+        from rich.table import Table
+
+        if console is None:
+            console = Console()
+
+        results_table = Table(title="Reliability Summary", show_header=True, header_style="bold magenta")
+        results_table.add_row("Evaluation Status", self.eval_status)
+        results_table.add_row("Failed Tool Calls", str(self.failed_tool_calls))
+        results_table.add_row("Passed Tool Calls", str(self.passed_tool_calls))
+        console.print(results_table)
+
+
+@dataclass
+class ReliabilityEvaluation:
+    """Evaluate the reliability of a model by checking the tool calls"""
+
+    # Evaluation name
+    name: Optional[str] = None
+    
+    # Evaluation UUID (autogenerated if not set)
+    eval_id: Optional[str] = None
+
+    # Evaluate an Agent
+    agent: Optional[Agent] = None
+
+    # Agent response
+    agent_response: Optional[RunResponse] = None
+
+    # Expected tool calls
+    expected_tool_calls: Optional[List[str]] = None
+
+    # Result of the evaluation
+    result: Optional[ReliabilityResult] = None
+
+    # Print summary of results
+    print_summary: bool = False
+    # Print detailed results
+    print_results: bool = False
+    # Save the result to a file
+    save_result_to_file: Optional[str] = None
+
+    # debug_mode=True enables debug logs
+    debug_mode: bool = False
+
+    def set_eval_id(self) -> str:
+        if self.eval_id is None:
+            self.eval_id = str(uuid4())
+        logger.debug(f"*********** Evaluation ID: {self.eval_id} ***********")
+        return self.eval_id
+
+    def set_debug_mode(self) -> None:
+        if self.debug_mode or getenv("AGNO_DEBUG", "false").lower() == "true":
+            self.debug_mode = True
+            set_log_level_to_debug()
+            logger.debug("Debug logs enabled")
+        else:
+            set_log_level_to_info()
+
+    def run(self, *, print_summary: bool = False, print_results: bool = False) -> None:
+        from rich.console import Console
+        from rich.live import Live
+        from rich.status import Status
+
+        self.set_eval_id()
+        self.set_debug_mode()
+        self.print_results = print_results
+        self.print_summary = print_summary
+
+        # Add a spinner while running the evaluations
+        console = Console()
+        with Live(console=console, transient=True) as live_log:
+            status = Status(f"Running evaluation...", spinner="dots", speed=1.0, refresh_per_second=10)
+            live_log.update(status)
+
+            last_tool_calls = None
```

Comment:
```suggestion
            actual_tool_calls = None
```
? Usually actual vs expected in tests